File: remove nth node from end of LL.py
# Time Complexity - O(n)
# Space Complexity - O(1)
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy = ListNode(-1)
        dummy.next = head
        s = dummy
        f = dummy
        for i in range(n):
            f = f.next
        while f.next!=None:
            s = s.next
            f = f.next
        s.next = s.next.next
        return dummy.next

File: RemoveNthNodeFromEndOfLL.java
// Time Complexity :O(n)
// Space Complexity :o(1)
// Did this code successfully run on Leetcode :Yes
// Any problem you faced while coding this :No


// Your code here along with comments explaining your approach: First run the fast loop till n and again run the fast and slow pointer with same speed to get element before the target.
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode slow= head;
        ListNode fast=head;

        for(int i=1; i<=n; i++){
            fast=fast.next;
        }
        if(fast==null){
            return head.next;
        }
        while(fast.next!=null){
            slow=slow.next;
            fast=fast.next;
        }
        slow.next=slow.next.next;
    return head;
    }
}